src/index.ts

```diff
@@ -107,15 +131,15 @@ export function createRouterLayout(
     async beforeRouteEnter(to, _from, next) {
       await loadAsyncComponents(to)
 
-      const name = resolveLayoutName(to.matched)
-      const layoutComp = name
-        ? normalizeEsModuleComponent(await resolve(name))
+      const layout = resolveLayout(to.matched)
+      const layoutComp = layout?.name
```

Comment:
```suggestion
      const layoutComp = layout
```

Reply:
I think this is a wrong suggestion, the `layout` is a normalized Layout object now for example `{name:string, props:{...}}`

Reply:
The point of the suggestion is we only need null-check for `layout` but not `layout.name`

Reply:
but then you are going to use `layout.name` right?

see:
https://github.com/yaquawa/vue-router-layout/blob/ff0931c6408412efb9ad18ebd99460c7d41dfd6d/src/index.ts#L142